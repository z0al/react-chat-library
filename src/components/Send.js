// @ts-check

// Packages
import React from 'react';

/**
 * @typedef Props
 * @property {string} [label]
 * @property {boolean} [disabled]
 * @property {boolean} [visible]
 * @property {object} [containerStyle]
 * @property {object} [buttonStyle]
 * @property {function} [onSubmit]
 * @param {Props} props
 */
const Send = props => (
	<div style={props.containerStyle}>
		<button
			type="button"
			onClick={() => props.onSubmit()}
			disabled={props.disabled}
			style={props.buttonStyle}
		>
			<span>{props.label}</span>
		</button>
		<style jsx>{`
			div {
				display: ${props.visible ? 'flex' : 'none'};
			}

			button {
				display: block;
				background: white;
				box-sizing: border-box;
				border: 1px solid gray;
				margin: 5px 0;
				padding: 0.3em 0.6em;
				border-radius: 5px;
			}

			span {
				display: block;
				outline: none;
				font-size: 1.2em;
			}
		`}</style>
	</div>
);

/** @type Props */
Send.defaultProps = {
	label: '+',
	disabled: false,
	visible: true,
	containerStyle: {},
	buttonStyle: {},
	onSubmit: () => {}
};

export default Send;
