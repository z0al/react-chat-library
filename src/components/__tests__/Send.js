import React from 'react';
import renderer from 'react-test-renderer';
import { render } from 'react-testing-library';

import Send from '../Send';

test('should match the snapshot', () => {
	const tree = renderer.create(<Send />).toJSON();
	expect(tree).toMatchSnapshot();
});

test('should show user custom label', () => {
	const label = 'Hello world';
	const { getByText } = render(<Send label={label} />);
	expect(getByText(label)).not.toBe(null);
});
