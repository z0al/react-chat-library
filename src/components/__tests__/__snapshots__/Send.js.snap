// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="jsx-2464195997"
  style={Object {}}
>
  <button
    className="jsx-2464195997"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <span
      className="jsx-2464195997"
    >
      +
    </span>
  </button>
</div>
`;
